xbmc/android/activity/XBMCApp.cpp

```diff
@@ -101,6 +102,7 @@ CXBMCApp::CXBMCApp(ANativeActivity* nativeActivity)
   : CJNIApplicationMainActivity(nativeActivity)
   , CJNIBroadcastReceiver("org/xbmc/kodi/XBMCBroadcastReceiver")
 {
+  m_xbmcappinstance = this;
```

Comment:
Oh, sure. This indeed 100% relies on the fact that we know that we have only 1 instance at a very specific place, and is completely invalid otherwise.
Anyway, I'll try to make it a proper singleton in 16.